layers/cmd/valueset.go

```diff
@@ -311,6 +319,103 @@ func LoadValuesetFiles(vs *Valuesets, files []string) error {
 	return nil
 }
 
+var re = regexp.MustCompile(`\([^()]*.?[^()]*\)$`)
+
+const (
+	sheetInputHeader  string = "input"
+	sheetOutputHeader string = "output"
+)
+
+func (vsets Valuesets) LoadSpreadsheets(ctx *ls.Context) error {
+	for _, spreadsheet := range vsets.Spreadsheets {
+		sheets, err := cmdutil.ReadSpreadsheetFile(spreadsheet)
+		if err != nil {
+			return fmt.Errorf("While reading file: %s, %w", spreadsheet, err)
+		}
+		sheetHeaders := make(map[string]string)
+		determineHeaderType := func(str string) (string, string) {
+			str = strings.TrimSpace(str)
+			e := re.FindString(str) // "(     i )"
+			headerType := strings.Trim(e, "(")
+			headerType = strings.Trim(headerType, ")")
+			switch strings.ToLower(strings.TrimSpace(headerType)) {
+			case "i":
+				return str[:strings.Index(str, "(")], sheetInputHeader
+			case "o":
+				return str[:strings.Index(str, "(")], sheetOutputHeader
+			}
+			return "", ""
+		}
+		for sheetName, sheet := range sheets {
+			if _, exists := vsets.Sets[sheetName]; exists {
+				return fmt.Errorf("Value set %s already defined -- in file: %s", sheetName, sheet)
+			}
+			var headerIdx int
+			var headerFlag bool
+		ROWS:
+			for rowIdx, row := range sheet {
+				if len(row) == 0 {
+					continue
+				}
+				for colIdx, columnData := range row {
+					if !headerFlag {
+						_, headerType := determineHeaderType(columnData)
+						sheetHeaders[columnData] = headerType
+						headerIdx = rowIdx
+						if colIdx == len(row)-1 {
+							headerFlag = true
+						}
+						continue
+					}
+					columnHeader := sheet[headerIdx][colIdx]
+					if sheetHeaders[columnHeader] == sheetInputHeader || sheetHeaders[columnHeader] == sheetOutputHeader {
+						ioHeaderType(sheetHeaders, sheetName, columnHeader, rowIdx, colIdx, sheet, &vsets)
+					} else {
+						kvPairHeaderType(sheetName, columnHeader, rowIdx, colIdx, sheet, &vsets)
```

Comment:
When processing a sheet:
  Each row is a value of the value set
  Each cell of the row should be matched with the column header and put into the kv-map

e.g.

Header: a, b, c, d
Row: 1, 2, 3, 4

From this, you should build a Value which has
a:1, b:2, c:3, d:4 in the kv-map